From 8cb4f07ce0a29100fee20b685d2d027dd05c1eb7 Mon Sep 17 00:00:00 2001
From: Haoran Wang <Haoran.Wang@freescale.com>
Date: Tue, 3 Nov 2015 06:27:32 +0800
Subject: [PATCH] MA-7064 Enable BCM4339 on evk_6ul

Added DTS file configuration for Murata V1+EX
BCM4339 module on evk_6ul platform.

Signed-off-by: Haoran Wang <Haoran.Wang@freescale.com>
---
 arch/arm/boot/dts/imx6ul-14x14-evk.dts | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/arch/arm/boot/dts/imx6ul-14x14-evk.dts b/arch/arm/boot/dts/imx6ul-14x14-evk.dts
index 8ae0e42..f2ef1f2 100644
--- a/arch/arm/boot/dts/imx6ul-14x14-evk.dts
+++ b/arch/arm/boot/dts/imx6ul-14x14-evk.dts
@@ -57,6 +57,7 @@
 			regulator-max-microvolt = <3300000>;
 			gpio = <&gpio1 9 GPIO_ACTIVE_HIGH>;
 			enable-active-high;
+			regulator-always-on;
 		};
 
 		reg_gpio_dvfs: regulator-gpio {
@@ -70,8 +71,22 @@
 			gpios = <&gpio5 3 GPIO_ACTIVE_HIGH>;
 			states = <1300000 0x1 1400000 0x0>;
 		};
+        wlreg_on: fixedregulator@100 {
+            compatible = "regulator-fixed";
+            regulator-min-microvolt = <5000000>;
+            regulator-max-microvolt = <5000000>;
+            regulator-name = "wlreg_on";
+            gpio = <&gpio5 1 0>;
+            startup-delay-us = <100>;
+            enable-active-high;
+        };
 	};
 
+    bcmdhd_wlan_0: bcmdhd_wlan@0 {
+        compatible = "android,bcmdhd_wlan";
+        wlreg_on-supply = <&wlreg_on>;
+    };
+
 	sound {
 		compatible = "fsl,imx6ul-evk-wm8960",
 			   "fsl,imx-audio-wm8960";
@@ -317,6 +332,9 @@
 	pinctrl-1 = <&pinctrl_usdhc1_100mhz>;
 	pinctrl-2 = <&pinctrl_usdhc1_200mhz>;
 	cd-gpios = <&gpio1 19 0>;
+	no-1-8-v;
+	wifi-host;
+	pm-ignore-notify;
 	keep-power-in-suspend;
 	enable-sdio-wakeup;
 	vmmc-supply = <&reg_sd1_vmmc>;
@@ -567,6 +585,7 @@
 				MX6UL_PAD_SD1_DATA1__USDHC1_DATA1 0x17059
 				MX6UL_PAD_SD1_DATA2__USDHC1_DATA2 0x17059
 				MX6UL_PAD_SD1_DATA3__USDHC1_DATA3 0x17059
+				MX6UL_PAD_SNVS_TAMPER1__GPIO5_IO01 0x03029
 			>;
 		};
 
@@ -578,6 +597,7 @@
 				MX6UL_PAD_SD1_DATA1__USDHC1_DATA1 0x170b9
 				MX6UL_PAD_SD1_DATA2__USDHC1_DATA2 0x170b9
 				MX6UL_PAD_SD1_DATA3__USDHC1_DATA3 0x170b9
+				MX6UL_PAD_SNVS_TAMPER1__GPIO5_IO01 0x03029
 			>;
 		};
 
@@ -589,6 +609,7 @@
 				MX6UL_PAD_SD1_DATA1__USDHC1_DATA1 0x170f9
 				MX6UL_PAD_SD1_DATA2__USDHC1_DATA2 0x170f9
 				MX6UL_PAD_SD1_DATA3__USDHC1_DATA3 0x170f9
+				MX6UL_PAD_SNVS_TAMPER1__GPIO5_IO01 0x03029
 			>;
 		};
 
-- 
1.8.0

